compiler/semcall.nim

```diff
@@ -247,7 +247,13 @@ proc presentFailedCandidates(c: PContext, n: PNode, errors: CandidateErrors):
           candidates.add typeToString(got)
           candidates.addDeclaredLocMaybe(c.config, got)
           doAssert wanted != nil
-          if got != nil: effectProblem(wanted, got, candidates, c)
+          if got != nil:
```

Comment:
how about changing title of this PR to something like:
make type mismatch errors more informative for calling convention/pragma mismatches